tests/transforms/test_metric_tensor.py

```diff
@@ -254,6 +243,37 @@ def circuit(a, b, c):
         )
         assert np.allclose(g, np.diag(expected), atol=tol, rtol=0)
 
+    def test_evaluate_diag_metric_tensor_classical_processing(self, tol):
+        """Test that a diagonal metric tensor evaluates correctly
+        when the QNode includes classical processing."""
+        dev = qml.device("default.qubit", wires=2)
+
+        def circuit(a, b):
+            qml.RX(a[1], wires=0)
+            qml.RY(a[0], wires=0)
+            qml.CNOT(wires=[0, 1])
+            qml.PhaseShift(b, wires=1)

```

Comment:
There is no classical processing of the arguments in this QNode

Reply:
There is, but it's subtle. The classical processing function is

```
f: ([a0, a1], b) -> (a1, a0, b)
```

So the classical Jacobians will be a permutation matrix and an identity matrix:

```
classical_jacobian(circuit)(a, b) == ([[0, 1], [1, 0]], [[1]])
```

Reply:
Have added a comment here :)

Reply:
Wow I totally missed that! Thanks, the comment is helpful :100: